src/components/ProductList/ProductList.jsx

```diff
@@ -1,96 +1,122 @@
-                    result = true;
-                        break;
-                    }
-                }
-                return result;
-            }
-        if(selectedCategories.length) {
-        setProducts(mock3.results.filter(filterProductsByCategory))
-        }    
-    }, [selectedCategories])
-    const sidebarWrapperStyle = (category) => {
-        return selectedCategories.some((element)=> element === category.slugs[0])  
-            ? "is-active"
-            : "";
+    if (categoriesData.results) {
+      setCategories(categoriesData.results);
     }
-    return (
-        <ProductListContainer>
-            <SidebarWrapper>
-                {mockProducts.results.map((category) => {
-                    return <ol 
-                    key={category.data.name}
-                    className={sidebarWrapperStyle(category)} 
-                    onClick={() => clickHandler(category.slugs[0])}>
-                        {category.data.name}</ol>
-                })}; 
-            </SidebarWrapper>
-            {loading ? (
-                <Loader />
-            ) : (
-            <ProductCard>
-            {products.map((product) => {
-                const productDetail = product.data;
-                return <Card 
-                    key={productDetail.sku}
-                    style={{ backgroundImage:`url(${productDetail.mainimage.url})` }}>
-                    <CardText>
-                        <h4 className="prod-name">{productDetail.name}</h4>
-                        <p className="price" >${productDetail.price}</p>
-                        <p><small>{productDetail.category.slug}</small></p> 
-                    </CardText>
-                </Card>
-                })}
-                <PaginationList>
-                <a href=".">0</a>
-                <a href=".">1</a>
-                <a href=".">2</a>
-                <a href=".">3</a>
-                </PaginationList>
-            </ProductCard>
-            )}
-            
-        </ProductListContainer>
-    )
-};
+  }, [data, categoriesData, category]);
+
+  const clickHandler = (categoryName) => {
+    const indexToRemove = selectedCategories.indexOf(categoryName);
+    if (indexToRemove !== -1) {
+      setSelectedCategories((prevValue) => {
+        const newArr = [...prevValue];
+        newArr.splice(indexToRemove, 1);
+        return newArr;
+      });
+    } else {
+      setSelectedCategories((prevValue) => [...prevValue, categoryName]);
+    }
+  };
+  /* const filterProductsByCategory = (product) => {
+    let result = false;
+    for (let i = 0; i < selectedCategories.length; i++) {
+      if (selectedCategories[i] === product.data.category.slug) {
+        result = true;
+        break;
+      }
+    }
+    return result;
+  }; */
+
+  useEffect(() => {
+    if (selectedCategories.length) {
+      //setProducts(data.results.filter(filterProductsByCategory));
+    }
+  }, [selectedCategories]);
+  const sidebarWrapperStyle = (category) => {
```

Comment:
maybe this can be split on 2 functions to keep it clean

Reply:
I will do a refactor in which this code no longer connected, but the api